src/Adapter/Product/Image/Repository/ProductImageRepository.php

```diff
@@ -364,4 +392,54 @@ public function delete(Image $image): void
     {
         $this->deleteObjectModel($image, CannotDeleteProductImageException::class);
     }
+
+    /**
+     * @param ProductId $productId
+     *
+     * @return string
+     *
+     * @throws CoreException
+     */
+    public function getProductCoverUrl(ProductId $productId): string
+    {
+        $imageId = $this->getDefaultImageId($productId);
+
+        return $imageId ?
+            $this->productImagePathFactory->getPath($imageId, ProductImagePathFactory::DEFAULT_IMAGE_FORMAT) :
+            $this->productImagePathFactory->getNoImagePath(ProductImagePathFactory::IMAGE_TYPE_SMALL_DEFAULT);
+    }
+
+    /**
+     * @param CombinationId $combinationId
+     *
+     * @return string
+     *
+     * @throws CoreException
+     */
+    public function getCombinationCoverUrl(CombinationId $combinationId): string
+    {
+        $imageId = $this->getPreviewCombinationProduct($combinationId);
+
+        return $imageId ?
+            $this->productImagePathFactory->getPath($imageId, ProductImagePathFactory::DEFAULT_IMAGE_FORMAT) :
+            $this->productImagePathFactory->getNoImagePath(ProductImagePathFactory::IMAGE_TYPE_SMALL_DEFAULT);
```

Comment:
Oh, I forgot about this one. There is a subtility for combination images, when no one is associated it is considered that all of them are associated. So in this case you can fallback on using the product cover. You don't have the productId in the parameters but it can be retrieved based on the combination ID

Reply:
```suggestion
        $imageId = $this->getPreviewCombinationProduct($combinationId);

        if ($imageId) {
            return $this->productImagePathFactory->getPath($imageId, ProductImagePathFactory::DEFAULT_IMAGE_FORMAT);
        }
        
        $productId = $this->getProductIdByCombinationId($combinationId);
        
        return $this->getProductCoverUrl($productId);
```

Reply:
For the `getProductIdByCombinationId` I implemented a `CombinationRepository::getProductIdByCombinationId` method in one of my PR but it's still not merged. You can either copy it in your PR and integrate `CombinationRepository` in this service if you're motivated (I'll deal with the conflicts no problem).

Or you can implement a small method with custom query here, and it will be replaced later when my PR is merged (in this case please add a `@todo Replace with CombinationRepository::getProductIdByCombinationId when #27353 is merged`

Reply:
ProductId is not anymore used in this method so the CombinationRepository::getProductIdByCombinationId is not mandatory for the moment. Am I wrong ?

Reply:
modifications applied